## 2. 공장에서는 진공관 수명이 1만 시간이라고 주장하여 품질관리팀에서 12개 샘플을 뽑았음. 유의 수준 5%에서 부호 검정

2-1. 귀무가설, 연구가설 세우기

- H0 : 데이터의 중위수는 1만시간이다. 
- H1 : 데이터의 중위수는 1만시간이 아니다. 

2-2. 유효한 데이터의 개수는?

In [2]:
import pandas as pd
import numpy as np

data = pd.read_csv("problem2.csv")
data.head()

name  life span
0  sample1      10000
1  sample2       9000
2  sample3       9500
3  sample4      10000
4  sample5      10000

In [4]:
data.shape

(12, 2)

In [7]:
data['life span'].median()

10000.0

In [8]:
print('중위수와 동일한 값은 순위 부호 검정에서 불필요함')
print('불필요한 값 : ',data[data['life span'] == 10000].shape[0])

df_filter = data[data['life span'] != 10000]

중위수와 동일한 값은 순위 부호 검정에서 불필요함
불필요한 값 :  4


3. 검정통계량 및 연구가설 채택 여부를 작성하라

In [9]:
from scipy.stats import wilcoxon
wilcoxon(df_filter['life span'] - 10000)

WilcoxonResult(statistic=8.5, pvalue=0.1953125)

- p-value는 0.195로 유의 수준 5%에서 귀무가설을 채택한다. 따라서 데이터의 중위수는 1만시간이라고 판단한다.